src/sinks/splunk_hec/logs/encoder.rs

```diff
@@ -95,30 +60,40 @@ impl Encoder<Vec<HecProcessedEvent>> for HecLogsEncoder {
         input: Vec<HecProcessedEvent>,
         writer: &mut dyn std::io::Write,
     ) -> std::io::Result<usize> {
+        let mut encoder = self.encoder.clone();
         let encoded_input: Vec<u8> = input
             .into_iter()
-            .filter_map(|e| self.encode_event(e))
+            .filter_map(|processed_event| {
+                let mut event = Event::from(processed_event.event);
+                let metadata = processed_event.metadata;
+                self.transformer.transform(&mut event);
+
+                let hec_event = if matches!(encoder.serializer(), Serializer::Json(_)) {
+                    HecEvent::Json(event.into_log())
+                } else {
+                    let mut bytes = BytesMut::new();
+                    encoder.encode(event, &mut bytes).ok()?;
+                    HecEvent::Bytes(bytes.freeze())
+                };
```

Comment:
I'm wondering if we want to ship `Serializer::NativeJson` as JSON or JSON-encoded string here.  Do you have thoughts on that, @lukesteensen @jszwedko?

Reply:
We should just match whatever the `json` encoding was previously doing.

Reply:
> We should just match whatever the json encoding was previously doing.

We do that in the `Serializer::Json` case here. I was interested specifically in the `NativeJson` case which is supposed to represent our internal events, so may differ from the `Json` codec in the future

Reply:
Ah, yes, sorry, I should have been clearer. I think we should encode it the same way as we encode `Json` to send it to Splunk. In this case, it should encode into `HecEvent::Json()`.

Reply:
I moved that logic into `Serializer::supports_json`/`Serializer::encode_json`:

https://github.com/vectordotdev/vector/blob/fa304f628d71145182a57d58bff228ef086118d5/src/sinks/splunk_hec/logs/encoder.rs#L72-L84